src/components/RegisterPage.tsx

```diff
@@ -0,0 +1,137 @@
+import React, { useState } from "react";
+import styled from "styled-components";
+import CreateOrLoginComponent from "./UI/CreateOrLoginComponent";
+import Form from "./UI/Form";
+import Input from "./UI/Input";
+import Button from "./UI/Button";
+import { useInput } from "../hooks/useInput.hook";
+import { AuthRequest } from "../models/request/AuthRequest";
+import { useRegisterUserMutation } from "../services/AuthService";
+
+const Container = styled.div`
+  margin-top: 5rem;
+  max-width: 20rem;
+  width: 100%;
+  height: auto;
+  margin: 7rem auto;
+`;
+
+const RegisterPage = () => {
+  const email = useInput({
+    initialValue: "",
+    validations: { isEmail: true, isEmpty: true },
+  });
+  const password = useInput({
+    initialValue: "",
+    validations: { minLength: 4, maxLength: 32, isEmpty: true },
+  });
+  const userName = useInput({
+    initialValue: "",
+    validations: { isEmpty: false },
+  });
+
+  const [registerUser] = useRegisterUserMutation();
+
+  const request: AuthRequest = {
+    action: "",
+    email: "",
+    password: "",
+    userName: "",
+  };
+
+  const isAllInputValid = () => {
+    return (
+      !email.valid.isValidInput ||
+      !password.valid.isValidInput ||
+      !userName.valid.isValidInput
+    );
+  };
+
+  const errorMessageDiv = (error: string) => {
+    return <div style={{ color: "red", marginTop: "0.5rem" }}>{error}</div>;
+  };
+  const emailErrorsRender = () => {
+    return email.isDirty && email.valid.isEmpty ? (
+      errorMessageDiv("Email can't be empty")
+    ) : email.isDirty && email.valid.emailError ? (
+      errorMessageDiv(email.valid.emailError)
+    ) : (
+      <></>
+    );
+  };
+  const passwordErrorsRender = () => {
+    if (password.isDirty) {
+      if (password.valid.isEmpty) {
+        return errorMessageDiv("Password can't be empty");
+      }
+      if (password.valid.minLengthError) {
+        return errorMessageDiv("Password" + password.valid.minLengthError);
+      }
+      if (password.valid.maxLengthError) {
+        return errorMessageDiv("Password" + password.valid.maxLengthError);
+      }
+    }
+  };
+
+  const handleRegister = async (e: React.MouseEvent) => {
+    e.preventDefault();
+    console.log("click");
+    request.action = "register";
+    request.email = email.value;
+    request.password = password.value;
+    request.userName = userName.value;
+    console.log(request);
+    const response = await registerUser(request).unwrap();
+    console.log(response);
```

Comment:
it's good behaviour to redirect user to another page after register [you did if for login page 👍 ]
also it's good to hide `register`&&`login` btn's when user is logged in.